internal/controlplane/handlers_ruletype.go

```diff
@@ -160,6 +165,34 @@ func (s *Server) CreateRuleType(
 
 	projectID := entityCtx.Project.ID
 
+	// As of the time of this writing, Minder profiles and rules
+	// have a guidance that's less than 10kB long.
+	if len(crt.RuleType.Guidance) > 10*1<<10 {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"invalid rule type definition: guidance too long",
+		)
+	}
+
+	md := goldmark.New(
+		// GitHub Flavored Markdown
+		goldmark.WithExtensions(extension.GFM),
+		goldmark.WithParserOptions(
+			parser.WithAutoHeadingID(),
+		),
+		goldmark.WithRendererOptions(
+			html.WithHardWraps(),
+			html.WithXHTML(),
+		),
+	)
+	if err := md.Convert([]byte(crt.RuleType.Guidance), &bytes.Buffer{}); err != nil {
+		return nil, status.Errorf(
```

Comment:
Should this be a user-facing issue? We have a `UserVisibleError` wrapper that allows for this.